### ISSUE TITLE
`role="button"` should be operable using both Enter and Space

### ISSUE THREAD

**Issue by danallsop** [ScottLogic/jumpstart-Leaflet#288]
> **Migrated from:** https://github.com/Leaflet/Leaflet/issues/7369
> **Original author:** danallsop
> **Original created:** 2020-12-08T14:05:37Z
---

## Motivation

Leaflet controls use `role="button"`<sup>[note]</sup> which is the implicit ARIA role of the `<button>` element, but the standard `<button>` element also activates on <kbd>Space</kbd>, so it may be confusing to some users that Leaflet "buttons" only activate on <kbd>Enter</kbd>.

[note]: Except where `role="button"` is currently missing:
- <s>Popup's close button</s>: https://github.com/Leaflet/Leaflet/issues/7399
- <s>Interactive markers</s>: https://github.com/Leaflet/Leaflet/issues/7428
- <s>Layer control</s>: https://github.com/Leaflet/Leaflet/issues/7498
- Fullscreen control: https://github.com/Leaflet/Leaflet.fullscreen/issues/110

## Proposed solution

The <span title="Web Accessibility Initiative - Accessible Rich Internet Applications">WAI-ARIA</span> Authoring Practices [recommended keyboard interactions for buttons](https://www.w3.org/WAI/ARIA/apg/patterns/button/#keyboardinteraction):

> When the button has focus:
> 
>  - <kbd>Space</kbd>: Activates the button.
>  - <kbd>Enter</kbd>: Activates the button.

**Comment by danallsop** [comment]
## Comments (2 total)

---
> **Comment by:** danallsop **on** 2021-11-21T21:26:45Z

danallsop Tested with the zoom controls (which should already work) but they are not triggered by hitting `space`

Have I understand it correct, that this should be fixed with only adding `role='button'`?
---
> **Comment by:** danallsop **on** 2021-11-28T17:05:26Z

danallsop the zoom controls are already `role="button"`, <kbd>space</kbd> would have to be handled manually for `<a role="button">` but works by default for `<button>`, opened: https://github.com/Leaflet/Leaflet/issues/7821.
